### PR TITLE
Fix unaligned access in unsafe fn tests

### PR DESCRIPTION
**Pull request by Enet4**
This should make the tests pass with Miri.

### CODE REVIEW

File: tests/test_util/aligned_vec.rs [hunk 1]
```diff
@@ -23,6 +30,10 @@ fn aligned_vec<T>(bytes: &[u8]) -> Vec<u8> {
         vec_len
     };
 
+    // the following code allocates a `Vec<T>` and turns it into
+    // a `Vec<u8>`. Assuming that this vector will not be dropped
+    // in this state, reading bytes from vit is safe.
```
Comment (nabijaczleweli) [inline tests/test_util/aligned_vec.rs]: "it", je presume?

File: tests/util/mod.rs [hunk 2]
```diff
@@ -48,15 +48,50 @@ fn smoke_check_alignment_from_8() {
...
+    unsafe {
+        let vec = aligned_vec::<u32>(data);
+        assert_eq!((vec.as_ptr() as usize) % align_of::<u32>(), 0);
+        dealloc_aligned_vec::<u32>(vec);
+    }
 
-    assert_eq!(aligned_vec::<u16>([].as_ref()), vec![]);
-    assert_eq!(aligned_vec::<i32>([].as_ref()), vec![]);
-    assert_eq!(aligned_vec::<u64>([].as_ref()), vec![]);
+    unsafe {
```
Comment (nabijaczleweli) [inline tests/util/mod.rs]: Are these extractible to a helper funxion?

File: tests/test_util/aligned_vec.rs [hunk 3]
```diff
@@ -11,8 +9,17 @@
 /// // the vector's data is guaranteed to be aligned for access as a u32
 /// assert_eq!((vec.as_ptr() as usize) % align_of::<u32>(), 0);
 /// ```
+/// 
+/// # Safety
+/// 
+/// The resulting vector must then be dealloc'd with the function
```
Comment (nabijaczleweli) [inline tests/test_util/aligned_vec.rs]: "Deallocated" or "freed" maybe?

**Comment by Enet4** [comment]
Rebass'd

**Comment by nabijaczleweli** [comment]
Thanks, I hate it!

**Pull request merged**